sdk/servicebus/service-bus/test/common/environmentVariables.ts

```diff
@@ -0,0 +1,98 @@
+const isNode = !!process && !!process.version && !!process.versions && !!process.versions.node;
+
+export enum Constants {
+  SERVICEBUS_CONNECTION_STRING = "SERVICEBUS_CONNECTION_STRING",
+  QUEUE_NAME = "QUEUE_NAME",
+  QUEUE_NAME_NO_PARTITION = "QUEUE_NAME_NO_PARTITION",
+  QUEUE_NAME_SESSION = "QUEUE_NAME_SESSION",
+  QUEUE_NAME_NO_PARTITION_SESSION = "QUEUE_NAME_NO_PARTITION_SESSION",
+  TOPIC_NAME = "TOPIC_NAME",
+  TOPIC_NAME_NO_PARTITION = "TOPIC_NAME_NO_PARTITION",
+  TOPIC_NAME_SESSION = "TOPIC_NAME_SESSION",
+  TOPIC_NAME_NO_PARTITION_SESSION = "TOPIC_NAME_NO_PARTITION_SESSION",
+  SUBSCRIPTION_NAME = "SUBSCRIPTION_NAME",
+  SUBSCRIPTION_NAME_NO_PARTITION = "SUBSCRIPTION_NAME_NO_PARTITION",
+  SUBSCRIPTION_NAME_SESSION = "SUBSCRIPTION_NAME_SESSION",
+  SUBSCRIPTION_NAME_NO_PARTITION_SESSION = "SUBSCRIPTION_NAME_NO_PARTITION_SESSION",
+  TOPIC_FILTER_NAME = "TOPIC_FILTER_NAME",
+  TOPIC_FILTER_SUBSCRIPTION_NAME = "TOPIC_FILTER_SUBSCRIPTION_NAME",
+  TOPIC_FILTER_DEFAULT_SUBSCRIPTION_NAME = "TOPIC_FILTER_DEFAULT_SUBSCRIPTION_NAME",
+  AAD_CLIENT_ID = "AAD_CLIENT_ID",
+  AAD_CLIENT_SECRET = "AAD_CLIENT_SECRET",
+  AAD_TENANT_ID = "AAD_TENANT_ID",
+  RESOURCE_GROUP = "RESOURCE_GROUP",
+  AZURE_SUBSCRIPTION_ID = "AZURE_SUBSCRIPTION_ID",
+  CLEAN_NAMESPACE = "CLEAN_NAMESPACE"
+}
+
+function _getMissingEnvironmentVariableError(envVar: any): Error {
+  return new Error(`Define ${envVar} in your environment before running integration tests.`);
+}
+
+function getEnv(name: string): string | undefined {
+  // @ts-ignore
+  const envValue = isNode ? process.env[name] : window.__env__[name];
+
+  if (!envValue) {
+    const mandatoryEnvVars = [Constants.SERVICEBUS_CONNECTION_STRING.valueOf()];
+
+    const aadRelatedEnvVars = [
```

Comment:
Since the `mandatoryEnvVars` and `aadRelatedEnvVars` are not changing with each call to `getEnv()`, we can pull it out of the function.

Reply:
Thanks! Ideally getEnv could do as little as possible. Catching missing required variables could be done only once before or after the environment variables are assigned.

Reply:
Yes that makes sense, will update in next round of changes.